### ISSUE TITLE
ENH:  add `atol`, `rtol` and `check_exact` to the Object.compare() method

### ISSUE THREAD

**Issue by tomhoq** [pandas-dev/pandas#58827]
### Feature Type

- [X] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

When comparing two different DataFrames or Series composed of columns of floats it can be quite useful to compare with an absolute tolerance (atol) and a relative tolerance(rtol) as we see in [pandas.testing.assert_frame_equal](https://pandas.pydata.org/docs/reference/api/pandas.testing.assert_frame_equal.html).

Unlike assert_frame_equal you would however, be able to pass dicts and lists to `atol` and `rtol `(in the DataFrame version of compare only), allowing you to specify a tolerance for each float column, or even having no tolerance for specific columns.

This is, particularly useful for DataFrames where there is a big discrepancy between columns and between values of the same column (where rtol would not serve). 

For instance if one column is in the magnitude of the `1*10^6` and another is on `1*10^-3`, a constant absolute tolerance for all columns is not ideal.

### Feature Description

Me and a colleague have already implemented the atol and rtol logic (without allowing dicts and lists being passed for now) by manually creating the mask that decides whether 2 elements in the same position should be considered equal or different.

As it was:
```
mask = ~(self == other | (self.isna() & other.isna()))
mask.fillna(True, inplace=True)
```
Our solution so far

```
if (not check_exact):
            if isinstance(self, ABCDataFrame):
                
                mask = np.ones(self.shape, dtype=bool)
                for col in self.columns:
                    if is_float_dtype(self[col]) and is_float_dtype(other[col]):
                        mask[:, self.columns.get_loc(col)] = np.isclose(self[col], other[col], rtol=rtol, atol=atol)
                    else:
                        mask[:, self.columns.get_loc(col)] = self[col] == other[col]
            #is series
            else: 
                if is_float_dtype(self):
                    mask = np.isclose(self, other, rtol=rtol, atol=atol)
                else:
                    mask = self == other
        else:
            mask = (self == other)
                
        mask = ~(mask | (self.isna() & other.isna()))
        mask.fillna(True, inplace=True)
```

### Alternative Solutions

NA

### Additional Context

A similar, less complete feature has already been proposed [here](https://github.com/pandas-dev/pandas/issues/54677) but has not yet been triaged.

**Comment by PushpitSB** [comment]
Yeah, this will really help while data selection for ML models, increasing the accuracy

**Comment by rhshadrach** [comment]
Thanks for the request. Closing as a duplicate of #48488

> A similar, less complete feature has already been proposed

Having multiple issues about what is really the same request is difficult from a maintenance standpoint. Instead of opening a new issue, it's preferred you comment and add your more complete version to the existing one.
